### PR TITLE
Fix #8464: Crash on game shutdown.

### LINKED ISSUE #8464: Client connection not cleaned up.
Currently  when the client disconnects or fails to connect it will call Network::CloseConnection which will delete server_connection and it is only ever created once in Init, subsequent attempts to connect to a server will eventually crash.

### PR DESCRIPTION
**Pull request by ZehMatt**
This PR fixes two issues, one is #8464 and the other one would be the network trying to access localisation on a null context.

### CODE REVIEW

File: src/openrct2/network/Network.cpp [hunk 1]
```diff
@@ -392,9 +392,6 @@ Network::Network()
 
 Network::~Network()
 {
-    CloseChatLog();
-    CloseServerLog();
-    CloseConnection();
 }
```
Comment (janisozaur) [inline src/openrct2/network/Network.cpp]: This has no purpose now, remove it from here and the class declaration as well.

**Comment by Broxzier** [comment]
This may fix #8021 as well.

**Comment by janisozaur** [comment]
@Broxzier yes it should fix that as well as great chunk of other issues reported by backtrace, like https://github.com/OpenRCT2/OpenRCT2/issues/8232 https://github.com/OpenRCT2/OpenRCT2/issues/8423 etc

**Comment by ZehMatt** [comment]
This is something I can not prove actually, I was unable to getting the game stuck on shutdown.

**Comment by janisozaur** [comment]
I'll have to comb through them and mark them fixed manually. :weary:

**Comment by ZehMatt** [comment]
> @Broxzier yes it should fix that as well as great chunk of other issues reported by backtrace, like #8232 #8423 etc

I think the mentioned issues are something else, they are not accessing the context, one of the issues rather seem to be related to object loading.

**Comment by janisozaur** [comment]
They will all try using `language_get_string` which is used somewhere in the destructors when it's already gone through network… I used to remember a more exact trace, but can't provide one right now.

**Comment by ZehMatt** [comment]
I really don't believe they are related after a closer.

**Pull request merged**